index.js

```diff
@@ -636,7 +636,12 @@ ShareDbMongo.prototype._getSnapshotOpLinkBulk = function(collectionName, ids, ca
 // **** Query methods
 
 ShareDbMongo.prototype._query = function(collection, inputQuery, projection, callback) {
-  var query = normalizeQuery(inputQuery);
+  try {
+    var query = this._normalizeQuery(inputQuery);
+  } catch (err) {
+    // XXX should we also print the stack trace?
+    callback({code: 5104, message: err.message});
```

Comment:
Should just add code to error object and pass it back. We'll sanitize before passing back to client in sharedb/agaent